app/src/main/java/mobi/inthepocket/android/beacons/app/service/BeaconActivityService.java

```diff
@@ -0,0 +1,79 @@
+import android.util.Log;
+import android.widget.Toast;
+
+import mobi.inthepocket.android.beacons.app.R;
+import mobi.inthepocket.android.beacons.ibeaconscanner.Beacon;
+import mobi.inthepocket.android.beacons.ibeaconscanner.BluetoothScanBroadcastReceiver;
+
+/**
+ * Target service responsible for receiving beacon updates from the library (and communicating them).
+ */
+public class BeaconActivityService extends JobIntentService
+{
+    private static final String TAG = "BeaconActivityService";
+
+    private static final int NOTIFICATION_ID = 65;
+
+    /**
+     * @see JobIntentService#onHandleWork(Intent)
+     */
+    @Override
+    protected void onHandleWork(@NonNull final Intent intent)
+    {
+        final Beacon beacon = intent.getParcelableExtra(BluetoothScanBroadcastReceiver.IBEACON_SCAN_BEACON_DETECTION);
+        final boolean enteredBeacon = intent.getBooleanExtra(BluetoothScanBroadcastReceiver.IBEACON_SCAN_BEACON_ENTERED, false);
+        final boolean exitedBeacon = intent.getBooleanExtra(BluetoothScanBroadcastReceiver.IBEACON_SCAN_BEACON_EXITED, false);
+
+        if (beacon != null)
+        {
+            String logMessage = "";
+            if (enteredBeacon)
+            {
+                Log.d(TAG, "entered beacon " + beacon.getUUID());
+                logMessage = String.format("Entered beacon with UUID %s and major %s and minor %s.", beacon.getUUID(), beacon.getMajor(), beacon.getMinor());
+            }
+            else if (exitedBeacon)
+            {
+                Log.d(TAG, "exited beacon " + beacon.getUUID());
+                logMessage = String.format("Exited beacon with UUID %s and major %s and minor %s.", beacon.getUUID(), beacon.getMajor(), beacon.getMinor());
+            }
+
+            // Create notification channel if required
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O)
+            {
+                final NotificationManager notificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
+                notificationManager.createNotificationChannel(new NotificationChannel(TAG, "Beacon Activity", NotificationManager.IMPORTANCE_LOW));
+            }
+
+            final Notification notification = new NotificationCompat.Builder(this, TAG)
+                    .setAutoCancel(true)
+                    .setContentText(logMessage)
+                    .setContentTitle("Beacon activity")
+                    .setGroup(TAG)
+                    .setPriority(NotificationCompat.PRIORITY_LOW)
+                    .setSmallIcon(R.mipmap.ic_launcher)
+                    .setStyle(new NotificationCompat.BigTextStyle().bigText(logMessage))
+                    .build();
+
+            final NotificationManagerCompat notificationManagerCompat = NotificationManagerCompat.from(this);
+            notificationManagerCompat.notify(TAG, NOTIFICATION_ID, notification);
+        }
+        else
+        {
+            // TODO: 14/03/2018 Add error support
```

Comment:
Anything more we can do here? if not, remove this todo

Reply:
Error handling was the "only" to do left for this PR.